### DISCUSSION TITLE
Add Z3950 and OpenLibrary

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#125] [General]
misspelled names are imported with OpenLibrary
exemples
9782259203043 Le Coeur a 2 places par Françoise Dorin
9782221091319 IL ETAIT MINUIT CINQ A BHOPAL

names well spelled with 
[Z39.50-1]
database_name = abes-z39-public
format = "utf-8"
hostname = carmin.sudoc.abes.fr
name = Sudoc 
password =
port = 10646
proxy_host =
proxy_port =
record_syntax = unimarc
username =

the 2nd one is the "Bibliothèque Nationale de France - BNF"  but I don't know if we can use it without authorization, I'm asking for information.

with OpenLibrary probably a small encoding problem

[test-add-Z3950-and-OpenLibrary.sqlite.tar.gz](https://github.com/textbrowser/biblioteq/files/7521902/test-add-Z3950-and-OpenLibrary.sqlite.tar.gz)

good courage :-)

**Comment by meteos77** [comment]
I did some tests and here is my feedback

[report_with_OpenLibrary_queries.txt](https://github.com/textbrowser/biblioteq/files/7669399/report_with_OpenLibrary_queries.txt)

[report_with_Zz3950-BnF_queries.txt](https://github.com/textbrowser/biblioteq/files/7669401/report_with_Zz3950-BnF_queries.txt)

**Reply by textbrowser** [reply]
https://en.wikipedia.org/wiki/Z39.50

The syntax may be because of the limitations of computers in the 1970s.

**Reply by meteos77** [reply]
Libraries are like ocean liners, changes are made over a long period of time, so a change represents a lot of work, so the international standards ...

**Reply by meteos77** [reply]
if I have followed everything correctly we went from Z3950 with marc; unimarc ...
to SRU with marcXML

**Reply by textbrowser** [reply]
> 

That's right. :)

**Comment by textbrowser** [comment]
If you can query the Z39.50 server openly, we can assume that you don't need permission. :)

**Reply by meteos77** [reply]
I have progressed since 1 month :-)

**Comment by textbrowser** [comment]
Submitting good information is a gift. You're doing well.

**Reply by meteos77** [reply]
the improvement of the software is above all for the users
so as a user trying to help you with our possibilities seems normal to me we are in the world of free software

the accented letters do not concern you I believe so if the languages which use them do not give you this information ...

**Comment by textbrowser** [comment]
I'll try removing the enumerations in the biblioteq.conf file. For example [Z39.50-1] to [Z39.50].

**Reply by meteos77** [reply]
if you work in this part of the program

in these Z3950 options if I put in the .conf file
the real name = catalogue g**é**n**é**ral de la BnF the display in the application the **"é**" become 

![accented-letter](https://user-images.githubusercontent.com/56431039/145077311-dbcdee00-2a35-42ad-95a4-c67a5bd920cd.png)

well it's not very important I change the "é" by "e  name = general catalog of the BnF  ... but it's not the right spelling anymore

**Reply by meteos77** [reply]
I don't find it problematic to list with xxx-1 xxx-2 but I'm waiting for what you will find better

**Reply by meteos77** [reply]
I did not understand the interest of SRU compared to Z3950 which seems to me more complete but here is an information to test a SRU which does not work :-)

SRU request but returns nothing while with firefox it's ok.
isbn-13 = 9782330048815

https://catalogue.bnf.fr/api/SRU?version=1.2&operation=searchRetrieve&query=bib.isbn%20adj%20%229782330048815%22

**Reply by textbrowser** [reply]
I suspect that the strings are misinterpreted. That's one of the difficult things with languages. A proper solution is to record the data as UTF-8. However, one has to have an editor which allows for UTF-8. Or, understand UTF-8.

**Reply by textbrowser** [reply]
SRU is much better than Z39.50. Z39.50 requires an entire infrastructure. SRU can be queried by almost anything.

**Reply by textbrowser** [reply]
SRU may more likely be successful with respect to firewalls as it works with port 80 and/or 443. It doesn't require credentials. It can work with proxies. It doesn't require Z39.50 libraries.

**Reply by meteos77** [reply]
conversely, it is UTF-8 that causes problems.
I saved the biblioteq.conf in Western (ISO-8859-15) and the **é** appeared
solved problem

**Reply by meteos77** [reply]
this is an impression but isn't there more information on the documents by Z3950 than by SRU?

**Reply by textbrowser** [reply]
> conversely, it is UTF-8 that causes problems. I saved the biblioteq.conf in Western (ISO-8859-15) and the e appeared solved problem

Perhaps the reader expects Latin-1. I'll know more when I look.

**Reply by textbrowser** [reply]
> this is an impression but isn't there more information on the documents by Z3950 than by SRU?

I don't know. :)

**Reply by meteos77** [reply]
no no it was an impression : I just checked between the Z3950 of BiblioteQ and the file retrieved by firefox from SRU, there are the same fields with the same information

**Reply by textbrowser** [reply]
SRU is XML? If so, it can provide the same information as Z39.50.

**Reply by textbrowser** [reply]
> catalogue g**é**n**é**ral de la BnF

Please test the recent change. The codec has been set to UTF-8 when accessing the INI (biblioteq.conf) file.

**Reply by meteos77** [reply]
cool it works

**Comment by meteos77** [comment]
is it easier for you to manage xml rather than unimarc?

**Reply by textbrowser** [reply]
The easiest has always been direct SQL. Zero parsing.

**Reply by meteos77** [reply]
I would ask the person at the departmental library if the file they provide us with when they lend us books can be in another format than unimarc with zone 995, you never know, that they use sql (even if I don't see how it works because the fields in their database are surely not the same as biblioteQ?)

**Reply by textbrowser** [reply]
SQL does not require parsing. You simply need to have some definitions on the schema. Then you can do whatever you wish with the data.

Schema design is difficult and I'm sure BQ's schema is not perfect.

**Reply by textbrowser** [reply]
With newer PostgreSQL engines, BQ supports row-level security. It's a very good feature. This means that a patron cannot read another patron's information (history, reserved items, etc.). Even if BQ prevented this, it ultimately falls on the database and PQ has this feature.

**Reply by textbrowser** [reply]
https://www.postgresql.org/docs/current/ddl-rowsecurity.html

**Comment by textbrowser** [comment]
SQLite, for example, requires one to enable foreign-key support at runtime.

https://www.sqlite.org/foreignkeys.html

PostgreSQL is much better with this.

**Comment by textbrowser** [comment]
BQ does not enable the support (via PRAGMA) because it's aware of the relationships. Even if it did enable it, people familiar with SQLite could violate the data by editing it through SQLite. With PostgreSQL, this is nearly impossible.

**Comment by textbrowser** [comment]
xbook_db=# select max(item_oid) from book_copy_info;
 max 
-----
 143
(1 row)

xbook_db=# update book_copy_info set item_oid = 150 where item_oid = 143;
ERROR:  insert or update on table "book_copy_info" violates foreign key constraint "book_copy_info_item_oid_fkey"
DETAIL:  Key (item_oid)=(150) is not present in table "book".

**Comment by meteos77** [comment]
it's sure that postgresql has more functionalities than Sqlite but they are not as old as Sqlite, see the additions that Sqlite will have in the future
If BiblioteQ hadn't proposed sqlite as a data support, I wouldn't have invested in it :-)
BiblioteQ with sqlite allows the transparency of the database just a file and it is GREAT in spite of the lesser functionalities.

**Comment by textbrowser** [comment]
SQLite was a request. I think.

April 12, 2008.

    Released version 4.02.
    SQLite 3.x is now supported.

**Reply by meteos77** [reply]
excellent idea in 2008 :-)
